Create my council account

ssrome opened the pull request:
- Added the local host as a base url in Cypress config so only have to put something like "/" or "/my-council-account/" in the test to get to page
- Tests for certain elements/text on the page
- Page with both buttons. Need to add the page name of where it should go
- Added an extra instruction to readme, for running tests - in separate PR

<img width="1493" alt="my-council-account" src="https://user-images.githubusercontent.com/17614754/182618510-a0dba301-4036-4b4d-a76e-1facd3b28fdf.png">

ssrome committed:
[Title goes here]

Summary:

Tasks:

ssrome committed:
[Title goes here]

Summary:

Tasks:

ssrome committed:
[update readme]

Summary: Change text on readme

Tasks:

ssrome committed:
Add components and tests to my council account page

Summary:
Also added local host address in cypress config

Tasks:

ssrome committed:
Add components and tests to my council account page

Summary:
Also added local host address in cypress config

Tasks:

ssrome committed:
Add onclick for button and test

Summary:

Tasks: Add correct url link for next page

ssrome committed:
Add onclick for button and test

Summary:

Tasks: Add correct url link for next page

ssrome committed:
Removed commented out line

Summary:

Tasks:

ssrome committed:
Removed commented out line

Summary:

Tasks:

ssrome committed:
Update header style

Summary:

Tasks:

ssrome committed:
Update header style

Summary:

Tasks:

ssrome committed:
Update header content

Summary:

Tasks:

ssrome committed:
Update header content

Summary:

Tasks:

ssrome committed:
Fix header on template file

Summary:

Tasks:

ssrome committed:
Change button alignment and text

Summary: Updated tests and added test for back link

Tasks:

ssrome committed:
Change button alignment and text

Summary: Updated tests and added test for back link

Tasks:

Joseaie commented on cypress.config.js:
```diff
@@ -2,6 +2,7 @@ const { defineConfig } = require("cypress");
 
 module.exports = defineConfig({
   e2e: {
+    baseUrl: "http://localhost:3000",
```
Nice!
nit: But technically should be part of a separate PR. 👀

Joseaie commented on cypress/e2e/spec.cy.js:
```diff
@@ -1,19 +1,19 @@
-    cy.contains('GOV.UK')
-  })
-})
+describe("empty spec", () => {
```
nit: these changes should be in a separate PR

Joseaie commented on readme.md:
```diff
@@ -35,7 +35,9 @@ Use `npm run devstart` to run server locally. This is so you don't have to resta
 ## Run Cypress tests locally
 
-Use `npm run e2e`
+Use `npm run e2e`. This will start the server, run the tests and close the server
```
nit: these changes should be in a separate PR

Joseaie commented on templates/default-page-template.html:
```diff
@@ -1,99 +1,173 @@
 <!DOCTYPE html>
-<html lang="en" class="govuk-template ">
```
nit: these changes should be in a separate PR

Joseaie commented on public/my-council-account/index.html:
```diff
@@ -0,0 +1,232 @@
+          </button>
+        </div>
+
+        <!-- <div class="govuk-body govuk-button-group">
```
nit: do we need the commented out bits?

ssrome replied:
Now deleted this

Joseaie commented:
I've left some comments on sections that should be part of different PRs as they don't relate to the title of this one (Single Responsibility and dat (even though that refers to code)). Other than that, I think your tests look pretty good, though I'm not sure if you wanted to update them with the `data-*` attributes.

Well done ✨

ssrome commented:
Gonna spilt these changes into separate PRs

ssrome committed:
Update tests to include data attributes

Summary:

Tasks:

ssrome committed:
Update tests to include data attributes

Summary:

Tasks:

ssrome committed:
Revert changes to pages

Summary:

Tasks:

ssrome committed:
Revert changes to pages

Summary:

Tasks:

ssrome commented:
The other pages that are showing as changed, are formatting things that I get fix. The actual changes for those files are in the separate PRs. This one should just be my tests and my council account page

ssrome committed:
Remove commented out buttons

Summary:

Tasks:

ssrome committed:
Remove commented out buttons

Summary:

Tasks:

ssrome committed:
Update cypress config (#13)

Summary:

Tasks:

Porim committed:
Ft dob (#9)

* Create a date of birth page with some basic tests

Summary: ^^^

Tasks:

* forgoten tests...

* add some new lines

* fix bug with header

* remove cancel button and add back button

Porim committed:
Delete .DS_Store

removing file that shouldn't be on github

Porim committed:
Delete .DS_Store

removing file that shouldn't be on github

Joseaie committed:
Create 'Name Page' (#18)

Summary: This is the page where the users enter their first and last
names. I have tested the buttons, inputs and labels.

Tasks: The buttons need to be linked to the real pages; for now they are
just hashes (`#`)

ssrome committed:
Add extra readme instruction, for running tests (#14)

Summary: Add extra readme instruction, for running tests

ssrome committed:
Add header with service name (#16)

Summary: Updated the header according to design system, that includes styling for the service name

ssrome committed:
Update page template header. Remove duplicate file (#15)

Removed the duplicate file in the public folder and updated the one in the templates folder, with the council name in the header. Rebased from main

ssrome committed:
Rebase branch from main
